Mathlib/Topology/UnitInterval.lean

```diff
@@ -179,6 +194,88 @@ theorem two_mul_sub_one_mem_iff {t : ℝ} : 2 * t - 1 ∈ I ↔ t ∈ Set.Icc (1
 
 end unitInterval
 
+section partition
+
+namespace Set.Icc
+
+variable {α} [LinearOrderedAddCommGroup α] {a b c d : α} (h : a ≤ b) {δ : α}
+
+-- TODO: Set.projIci, Set.projIic
+/-- `Set.projIcc` is a contraction. -/
+lemma _root_.Set.abs_projIcc_sub_projIcc : (|projIcc a b h c - projIcc a b h d| : α) ≤ |c - d| := by
+  wlog hdc : d ≤ c generalizing c d
+  · rw [abs_sub_comm, abs_sub_comm c]; exact this (le_of_not_le hdc)
+  rw [abs_eq_self.2 (sub_nonneg.2 hdc), abs_eq_self.2 (sub_nonneg.2 <| monotone_projIcc h hdc)]
+  rw [← sub_nonneg] at hdc
+  refine (max_sub_max_le_max _ _ _ _).trans (max_le (by rwa [sub_self]) ?_)
+  refine ((le_abs_self _).trans <| abs_min_sub_min_le_max _ _ _ _).trans (max_le ?_ ?_)
+  · rwa [sub_self, abs_zero]
+  · exact (abs_eq_self.mpr hdc).le
+
+/-- When `h : a ≤ b` and `δ > 0`, `addNsmul h δ` is a sequence of points in the closed interval
+  `[a,b]`, which is initially equally spaced but eventually stays at the right endpoint `b`. -/
+def addNsmul (δ : α) (n : ℕ) : Icc a b := projIcc a b h (a + n • δ)
+
+lemma addNsmul_zero : addNsmul h δ 0 = a := by
+  rw [addNsmul, zero_smul, add_zero, projIcc_left]
+
+lemma addNsmul_eq_right [Archimedean α] (hδ : 0 < δ) :
+    ∃ m, ∀ n ≥ m, addNsmul h δ n = b := by
+  obtain ⟨m, hm⟩ := Archimedean.arch (b - a) hδ
+  refine ⟨m, fun n hn ↦ ?_⟩
+  rw [addNsmul, coe_projIcc, add_comm, min_eq_left_iff.mpr, max_eq_right h]
+  exact sub_le_iff_le_add.mp (hm.trans <| nsmul_le_nsmul hδ.le hn)
+
+lemma monotone_addNsmul (hδ : 0 ≤ δ) : Monotone (addNsmul h δ) :=
+  fun _ _ hnm ↦ monotone_projIcc h <| (add_le_add_iff_left _).mpr (nsmul_le_nsmul hδ hnm)
+
+lemma abs_sub_addNsmul_le (hδ : 0 ≤ δ) {t : Icc a b} (n : ℕ)
+    (ht : t ∈ Icc (addNsmul h δ n) (addNsmul h δ (n+1))) :
+    (|t - addNsmul h δ n| : α) ≤ δ :=
+  (abs_eq_self.2 <| sub_nonneg.2 ht.1).trans_le <| (sub_le_sub_right (by exact ht.2) _).trans <|
+    (le_abs_self _).trans <| (abs_projIcc_sub_projIcc h).trans <| by
+      rw [add_sub_add_comm, sub_self, zero_add, succ_nsmul, add_sub_cancel]
+      exact (abs_eq_self.mpr hδ).le
+
+end Set.Icc
+
+open scoped unitInterval
+
+/-- Any open cover `c` of a closed interval `[a, b]` in ℝ can be refined to
+  a finite partition into subintervals. -/
+lemma lebesgue_number_lemma_Icc {ι} {a b : ℝ} (h : a ≤ b) {c : ι → Set (Icc a b)}
```

Comment:
I don't think is a great name. We usually don't name theorem using the main proof ingredient, we try to describe the statement. This also applies to the next two statements.

Reply:
Indeed; I had the impression it's roughly equivalent to the Lebesgue number lemma, but there's no obvious reverse implication. Changed to `exists_monotone_Icc_subset_open_cover_Icc/unitInterval(_prod_self)`.